engine/src/flutter/impeller/geometry/rect.h

```diff
@@ -485,6 +485,18 @@ struct TRect {
     FML_UNREACHABLE();
   }
 
+  /// @brief  Creates a new bounding box that contains this transformed
+  ///          rectangle.
+  ///
+  /// [transform] must be a translate-scale only matrix.
+  [[nodiscard]] constexpr TRect TransformBoundsTranslateScale(
+      const Matrix& transform) const {
+    return TRect::MakeOriginSize(
+        GetOrigin() + TPoint<float>(transform.m[12], transform.m[13]),  //
+        GetSize().Scale(transform.m[0], transform.m[5])                 //
```

Comment:
This is not correct. You have to scale all 4 points.

Also, if there is a negative scale it can mess things up - your rect will be upside down.

Reply:
Update the unit tests with lots of cases of rects at origin or not at origin, various sizes, and various scales and translates and make sure the answer is the same as the full transform.